`os_driver activating...` followed by `process has died`

When I am running the ouster_ros driver launch file in ros2-humble, it starts up normally connecting to the sensor and then in the terminal when it comes to activating the sensor it shuts down the driver node

```
[os_driver-2] [INFO] [1727995391.847437646] [ouster.os_driver]: Retrived sensor active config
[os_driver-2] [INFO] [1727995391.847511566] [ouster.os_driver]: Will send UDP data to 10.42.0.1
[os_driver-2] [INFO] [1727995391.847523008] [ouster.os_driver]: Contacting sensor 10.42.0.89 ...
[os_driver-2] [INFO] [1727995391.863511634] [ouster.os_driver]: Sensor 10.42.0.89 configured successfully
[os_driver-2] [INFO] [1727995391.863533340] [ouster.os_driver]: Starting sensor 10.42.0.89 initialization... Using ports: 7502/7503
[os_driver-2] [2024-10-03 22:43:11.863] [ouster::sensor] [info] initializing sensor client: 10.42.0.89 expecting lidar port/imu port: 7502/7503
[os_driver-2] [2024-10-03 22:43:11.937] [ouster::sensor] [info] parsing non-legacy metadata format
[os_driver-2] [INFO] [1727995391.938583205] [ouster.os_driver]: No metadata file was specified, using: 10.42.0-metadata.json
[os_driver-2] [INFO] [1727995391.938716769] [ouster.os_driver]: Wrote sensor metadata to 10.42.0-metadata.json
[os_driver-2] [INFO] [1727995391.938736095] [ouster.os_driver]: ouster client version: 0.11.1+483206f-release
[os_driver-2] product: OS-1-128, sn: 122426000526, firmware rev: v3.1.0
[os_driver-2] lidar mode: 1024x10, lidar udp profile: RNG19_RFL8_SIG16_NIR16
[os_driver-2] [INFO] [1727995391.939190116] [ouster.os_driver]: reset service created
[os_driver-2] [INFO] [1727995391.939500966] [ouster.os_driver]: get_metadata service created
[os_driver-2] [INFO] [1727995391.939704258] [ouster.os_driver]: get_config service created
[os_driver-2] [INFO] [1727995391.939912723] [ouster.os_driver]: set_config service created
[INFO] [launch.user]: os_driver activating...
[ERROR] [os_driver-2]: process has died [pid 9447, exit code -11, cmd '/docker_ros2_ws/install/ouster_ros/lib/ouster_ros/os_driver --ros-args -r __node:=os_driver -r __ns:=/ouster --params-file /docker_ros2_ws/install/ouster_ros/share/ouster_ros/config/driver_params.yaml'].
```

After colcon building my environment and `source install/setup.bash`, I simply run
`
ros2 launch ouster_ros driver.launch.py
`

I've edited the `driver_params.yaml` file with the correct IP address and set `use_system_default_qos: false`. Everything else is the same. I've done this before and it worked.

**System**
LiDAR: OS1-128
System: Ubuntu 22.04 in Docker

I get the same error with the new version

```
[os_driver-2] [INFO] [1728326550.047323450] [ouster.os_driver]: Will use automatic UDP destination
[os_driver-2] [INFO] [1728326550.047358267] [ouster.os_driver]: Contacting sensor 10.42.0.89 ...
[os_driver-2] [INFO] [1728326550.085492051] [ouster.os_driver]: Sensor 10.42.0.89 configured successfully
[os_driver-2] [INFO] [1728326550.085542937] [ouster.os_driver]: Starting sensor 10.42.0.89 initialization... Using ports: 0/0
[os_driver-2] [2024-10-07 18:42:30.085] [ouster::sensor] [info] initializing sensor client: 10.42.0.89 expecting lidar port/imu port: 0/0
[os_driver-2] [2024-10-07 18:42:30.085] [ouster::sensor] [info] (0 means a random port will be chosen)
[rviz2-1] [INFO] [1728326551.383241332] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-1] [INFO] [1728326551.383437922] [rviz2]: OpenGl version: 4.5 (GLSL 4.5)
[rviz2-1] [INFO] [1728326551.531269751] [rviz2]: Stereo is NOT SUPPORTED
[rviz2-1] [INFO] [1728326551.937348032] [rviz2]: Stereo is NOT SUPPORTED
[os_driver-2] [2024-10-07 18:42:42.382] [ouster::sensor] [info] parsing non-legacy metadata format
[os_driver-2] [INFO] [1728326562.387811145] [ouster.os_driver]: No metadata file was specified, using: 10.42.0-metadata.json
[os_driver-2] [INFO] [1728326562.388235065] [ouster.os_driver]: Wrote sensor metadata to 10.42.0-metadata.json
[os_driver-2] [INFO] [1728326562.388310183] [ouster.os_driver]: ouster client version: 0.11.1+483206f-release
[os_driver-2] product: OS-1-128, sn: 122426000526, firmware rev: v3.1.0
[os_driver-2] lidar mode: 1024x10, lidar udp profile: RNG19_RFL8_SIG16_NIR16
[os_driver-2] [INFO] [1728326562.403405172] [ouster.os_driver]: reset service created
[os_driver-2] [INFO] [1728326562.405358107] [ouster.os_driver]: get_metadata service created
[os_driver-2] [INFO] [1728326562.407910454] [ouster.os_driver]: get_config service created
[os_driver-2] [INFO] [1728326562.409081627] [ouster.os_driver]: set_config service created
[INFO] [launch.user]: os_driver activating...
[ERROR] [os_driver-2]: process has died [pid 1339, exit code -11, cmd '/docker_ros2_ws/install/ouster_ros/lib/ouster_ros/os_driver --ros-args -r __node:=os_driver -r __ns:=/ouster --params-file /docker_ros2_ws/install/ouster_ros/share/ouster_ros/config/driver_params.yaml'].
```

and the version I have is 0.13.1: `ros2 pkg xml ouster_ros --tag version 0.13.1`

I got the same error when setting `viz:=false`. It used to work in my docker container and display the live stream LiDAR data in ros2. It just suddenly stopped. I'm still getting the same error.

```
ros2 launch ouster_ros driver.launch.py viz:=false
[INFO] [launch]: All log files can be found below /root/.ros/log/2024-10-07-20-31-49-834607-CB490-EE11723-18877
[INFO] [launch]: Default logging verbosity is set to INFO
/docker_ros2_ws/install/ouster_ros/share/ouster_ros
/docker_ros2_ws/install/ouster_ros/share/ouster_ros/config/viz.rviz
[INFO] [os_driver-1]: process started with pid [18888]
[os_driver-1] [WARN] [1728333109.937227903] [ouster.os_driver]: lidar port set to zero, the client will assign a random port number!
[os_driver-1] [WARN] [1728333109.937292500] [ouster.os_driver]: imu port set to zero, the client will assign a random port number!
[os_driver-1] [INFO] [1728333110.165197522] [ouster.os_driver]: Will use automatic UDP destination
[os_driver-1] [INFO] [1728333110.165323389] [ouster.os_driver]: Contacting sensor 10.42.0.89 ...
[os_driver-1] [INFO] [1728333110.205681268] [ouster.os_driver]: Sensor 10.42.0.89 configured successfully
[os_driver-1] [INFO] [1728333110.205831471] [ouster.os_driver]: Starting sensor 10.42.0.89 initialization... Using ports: 0/0
[os_driver-1] [2024-10-07 20:31:50.205] [ouster::sensor] [info] initializing sensor client: 10.42.0.89 expecting lidar port/imu port: 0/0
[os_driver-1] [2024-10-07 20:31:50.206] [ouster::sensor] [info] (0 means a random port will be chosen)
[os_driver-1] [2024-10-07 20:32:02.492] [ouster::sensor] [info] parsing non-legacy metadata format
[os_driver-1] [INFO] [1728333122.495792256] [ouster.os_driver]: No metadata file was specified, using: 10.42.0-metadata.json
[os_driver-1] [INFO] [1728333122.495966478] [ouster.os_driver]: Wrote sensor metadata to 10.42.0-metadata.json
[os_driver-1] [INFO] [1728333122.495987888] [ouster.os_driver]: ouster client version: 0.11.1+483206f-release
[os_driver-1] product: OS-1-128, sn: 122426000526, firmware rev: v3.1.0
[os_driver-1] lidar mode: 1024x10, lidar udp profile: RNG19_RFL8_SIG16_NIR16
[os_driver-1] [INFO] [1728333122.496568409] [ouster.os_driver]: reset service created
[os_driver-1] [INFO] [1728333122.496927694] [ouster.os_driver]: get_metadata service created
[os_driver-1] [INFO] [1728333122.497161732] [ouster.os_driver]: get_config service created
[os_driver-1] [INFO] [1728333122.497411519] [ouster.os_driver]: set_config service created
[INFO] [launch.user]: os_driver activating...
[ERROR] [os_driver-1]: process has died [pid 18888, exit code -11, cmd '/docker_ros2_ws/install/ouster_ros/lib/ouster_ros/os_driver --ros-args -r __node:=os_driver -r __ns:=/ouster --params-file /docker_ros2_ws/install/ouster_ros/share/ouster_ros/config/driver_params.yaml'].
```

--- a/CHANGELOG.rst
+++ b/CHANGELOG.rst
@@ -2,9 +2,11 @@
 Changelog
 =========
 
-ouster_ros v0.13.1
+ouster_ros v0.13.2
 ==================
 * [BUGFIX]: Make sure to initialize the sensor with launch file parameters.
+* [BUGFIX]: ``os_driver`` failed when RAW option is used.
+
 
 ouster_ros v0.13.0
 ==================

--- a/ouster-ros/package.xml
+++ b/ouster-ros/package.xml
@@ -2,7 +2,7 @@
 <?xml-model href="http://download.ros.org/schema/package_format2.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
   <name>ouster_ros</name>
-  <version>0.13.1</version>
+  <version>0.13.2</version>
   <description>Ouster ROS2 driver</description>
   <maintainer email="oss@ouster.io">ouster developers</maintainer>
   <license file="LICENSE">BSD</license>

--- a/ouster-ros/src/os_driver_node.cpp
+++ b/ouster-ros/src/os_driver_node.cpp
@@ -200,12 +200,14 @@ class OusterDriver : public OusterSensor {
                 static_cast<int64_t>(ptp_utc_tai_offset * 1e+9));
 
         publish_raw = impl::check_token(tokens, "RAW");
+        if (publish_raw)
+            OusterSensor::create_publishers();
     }
 
     virtual void on_lidar_packet_msg(const LidarPacket& lidar_packet) override {
         if (lidar_packet_handler)
             lidar_packet_handler(lidar_packet);
-        
+
         if (publish_raw)
             OusterSensor::on_lidar_packet_msg(lidar_packet);
     }
@@ -213,8 +215,8 @@ class OusterDriver : public OusterSensor {
     virtual void on_imu_packet_msg(const ImuPacket& imu_packet) override {
         if (imu_packet_handler)
             imu_pub->publish(imu_packet_handler(imu_packet));
-        
-        if(publish_raw)
+
+        if (publish_raw)
             OusterSensor::on_imu_packet_msg(imu_packet);
     }
 

--- a/ouster-ros/src/os_pcap_node.cpp
+++ b/ouster-ros/src/os_pcap_node.cpp
@@ -75,8 +75,6 @@ class OusterPcap : public OusterSensorNodeBase {
         RCLCPP_DEBUG(get_logger(), "on_activate() is called.");
         LifecycleNode::on_activate(state);
         create_publishers();
-        if (imu_packet_pub) imu_packet_pub->on_activate();
-        if (lidar_packet_pub) lidar_packet_pub->on_activate();
         allocate_buffers();
         start_packet_read_thread();
         return LifecycleNodeInterface::CallbackReturn::SUCCESS;
@@ -278,10 +276,8 @@ class OusterPcap : public OusterSensorNodeBase {
     std::shared_ptr<PcapReader> pcap;
     ouster_sensor_msgs::msg::PacketMsg lidar_packet;
     ouster_sensor_msgs::msg::PacketMsg imu_packet;
-    rclcpp_lifecycle::LifecyclePublisher<ouster_sensor_msgs::msg::PacketMsg>::SharedPtr
-        lidar_packet_pub;
-    rclcpp_lifecycle::LifecyclePublisher<ouster_sensor_msgs::msg::PacketMsg>::SharedPtr
-        imu_packet_pub;
+    rclcpp::Publisher<ouster_sensor_msgs::msg::PacketMsg>::SharedPtr lidar_packet_pub;
+    rclcpp::Publisher<ouster_sensor_msgs::msg::PacketMsg>::SharedPtr imu_packet_pub;
 
     std::atomic<bool> packet_read_active = {false};
     std::unique_ptr<std::thread> packet_read_thread;

--- a/ouster-ros/src/os_sensor_node.cpp
+++ b/ouster-ros/src/os_sensor_node.cpp
@@ -769,11 +769,10 @@ void OusterSensor::create_publishers() {
 
 void OusterSensor::allocate_buffers() {
     auto& pf = sensor::get_format(info);
-
-    lidar_packet_msg.buf.resize(pf.lidar_packet_size);
     lidar_packet.buf.resize(pf.lidar_packet_size);
+    lidar_packet_msg.buf.resize(pf.lidar_packet_size);
     imu_packet.buf.resize(pf.imu_packet_size);
-    imu_packet.buf.resize(pf.imu_packet_size);
+    imu_packet_msg.buf.resize(pf.imu_packet_size);
 }
 
 bool OusterSensor::init_id_changed(const sensor::packet_format& pf,

